analyzers/src/SonarAnalyzer.CSharp/Rules/LoggingArgumentsShouldBePassedCorrectly.cs

```diff
@@ -88,7 +89,7 @@ private static void CheckInvalidParams(InvocationExpressionSyntax invocation, IM
 
     private static void CheckInvalidTypeParams(InvocationExpressionSyntax invocation, IMethodSymbol methodSymbol, SonarSyntaxNodeReportingContext c, ImmutableArray<KnownType> knownTypes)
     {
-        if (!IsNLogIgnoredOverload(methodSymbol) && methodSymbol.TypeArguments.Any(x => x.DerivesFromAny(knownTypes)))
+        if (!knownTypes.IsEmpty && !IsNLogIgnoredOverload(methodSymbol) && methodSymbol.TypeArguments.Any(x => x.DerivesFromAny(knownTypes)))
         {
             var typeParameterNames = methodSymbol.TypeParameters.Select(x => x.MetadataName).ToArray();
             var positions = methodSymbol.ConstructedFrom.Parameters.Where(x => typeParameterNames.Contains(x.Type.MetadataName)).Select(x => methodSymbol.ConstructedFrom.Parameters.IndexOf(x));
```

Comment:
Outside of this PR, but I think it's useful:

On `HasLoggingMethodName`, you do 4 `Contains` checks and you get the `invocation.GetName()` four times.
I would make a private static HashSet on this class like so:

```csharp
private static readonly HashSet<string> loggingMethodNames = MicrosoftExtensionsLogging
.Concat(NLogLoggingMethods)
.Concat(Serilog)
.Concat(CastleCoreOrCommonCore)
.ToHashSet();
```

...and do one `Contains`.